manifests/dashboard.pp

```diff
@@ -4,66 +4,108 @@
+  }
+
   # assign version according to the package manager
-  case $::osfamily {
-    'Debian' : {
+  case $facts['os']['family'] {
+    'Debian': {
       $dashboard_version_install = "${dashboard_version}-*"
     }
-    'Linux', 'RedHat' : {
-      $dashboard_version_install = "${dashboard_version}"
+    'Linux', 'RedHat', default: {
+      $dashboard_version_install = $dashboard_version
     }
   }
 
   # install package
-  package { 'Installing Wazuh Dashboard...':
+  package { 'wazuh-dashboard':
     ensure => $dashboard_version_install,
     name   => $dashboard_package,
   }
 
-  include wazuh::certificates
+  require wazuh::certificates
 
-  exec { 'Copy Dashboard Certificates':
+  exec { "ensure full path of ${dashboard_path_certs}":
     path    => '/usr/bin:/bin',
-    command => "mkdir $dashboard_path_certs \
-             && cp /tmp/wazuh-certificates/dashboard.pem  $dashboard_path_certs\
-             && cp /tmp/wazuh-certificates/dashboard-key.pem  $dashboard_path_certs\
-             && cp /tmp/wazuh-certificates/root-ca.pem  $dashboard_path_certs\
-             && chown wazuh-dashboard:wazuh-dashboard -R $dashboard_path_certs\
-             && chmod 500 $dashboard_path_certs\
-             && chmod 400 $dashboard_path_certs/*",
+    command => "mkdir -p ${dashboard_path_certs}",
+    creates => $dashboard_path_certs,
+    require => Package['wazuh-dashboard'],
+  }
+  -> file { $dashboard_path_certs:
+    ensure => directory,
+    owner  => $dashboard_fileuser,
+    group  => $dashboard_filegroup,
+    mode   => '0500',
+  }
 
+  [
+    'dashboard.pem',
+    'dashboard-key.pem',
+    'root-ca.pem',
+  ].each |String $certfile| {
+    file { "${dashboard_path_certs}/${certfile}":
+      ensure  => file,
+      owner   => $dashboard_fileuser,
+      group   => $dashboard_filegroup,
+      mode    => '0400',
+      replace => false,  # only copy content when file not exist
+      source  => "/tmp/wazuh-certificates/${certfile}",
+    }
+  }
+
+  # TODO: Fully manage the opensearch_dashboards.yml and a template file resource
+  file { '/etc/wazuh-dashboard/opensearch_dashboards.yml':
+    owner   => 'wazuh-dashboard',
+    group   => 'wazuh-dashboard',
+    mode    => '0640',
+    content => template('wazuh/opensearch_dashboards_yml.erb'),
+    require => Package[$dashboard_package],
+    notify  => Service[$dashboard_service]
   }
 
   service { 'wazuh-dashboard':
     ensure     => running,
     enable     => true,
     hasrestart => true,
+    name       => $dashboard_service,
   }
 
-  exec {'Waiting for Wazuh indexer...':
-    path      => '/usr/bin',
-    command   => "curl -u ${dashboard_user}:${dashboard_password} -k -s -XGET https://${indexer_server_ip}:${indexer_server_port}",
-    tries     => 100,
-    try_sleep => 3,
+  exec {'Waiting for Wazuh dashboard...':
+    require => Service[$dashboard_service],
+    command => "sleep 15 ",
+    path => "/usr/bin:/bin",
   }
```

Comment:
The reason for this sleep is because the file /usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml is created on the first start of Wazuh dashboard, so we must give it a delay time to make sure that the step file { '/usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml': can be executed successfully

Reply:
😅 ok what you want is that the file is manged before the service starts i think. Because you mange the content of the file. Else you will create a loop were puppet changes the file and when the dashboard starts it overrides it again.

I will add a suggestion in a second to this pull request. So you can see what i mean.